src/wallet/test/silentpayment_tests.cpp

```diff
@@ -0,0 +1,161 @@
+            }
+
+            BOOST_CHECK(silent_payment_addresses.size() == expected_spks.size());
+            for (const auto& spk : silent_payment_addresses) {
+                BOOST_CHECK(std::find(expected_spks.begin(), expected_spks.end(), spk) != expected_spks.end());
+            }
+        }
+
+        // Test receiving
+        for (const auto& recipient : vec["receiving"].getValues()) {
+            // TODO: implement labels for Bitcoin Core, until then skip the receiving with labels tests
+            if (recipient["supports_labels"].get_bool()) {
+                BOOST_TEST_MESSAGE("Labels not implemented; skipping..");
+                continue;
+            }
+
+            const auto& given = recipient["given"];
+            const auto& expected = recipient["expected"];
+
+            std::vector<COutPoint> outpoints;
+            for (const auto& outpoint : recipient["given"]["outpoints"].getValues()) {
+                outpoints.emplace_back(uint256S(outpoint[0].get_str()), outpoint[1].getInt<uint32_t>());
+            }
+
+            std::vector<CPubKey> input_pub_keys;
+            for (const auto& pubkey : given["input_pub_keys"].getValues()) {
+                // All pubkeys must be in compressed format
+                auto pubkey_bytes = ParseHex(pubkey.get_str());
+                if (pubkey_bytes.size() == 32) {
+                    // XOnlyPubKeys are always even
+                    pubkey_bytes.insert(pubkey_bytes.begin(), 2);
+                }
+                input_pub_keys.emplace_back(pubkey_bytes);
+            }
+            std::vector<XOnlyPubKey> output_pub_keys;
+            for (const auto& pubkey : given["outputs"].getValues()) {
+                output_pub_keys.emplace_back(ParseHex(pubkey.get_str()));
+            }
+
+            std::string hex_str = given["bip32_seed"].get_str();
+            std::vector<std::byte> seed{ParseHex<std::byte>(hex_str)};
+            std::vector<uint32_t> scan_keypath;
+            BOOST_CHECK(ParseHDKeypath("m/352'/0'/0'/1'/0", scan_keypath));
+            std::vector<uint32_t> spend_keypath;
+            BOOST_CHECK(ParseHDKeypath("m/352'/0'/0'/0'/0", spend_keypath));
+            CKey scan_priv_key = GetKeyFromBIP32Path(seed, scan_keypath);
+            CKey spend_priv_key = GetKeyFromBIP32Path(seed, spend_keypath);
+            CPubKey spend_pubkey = spend_priv_key.GetPubKey();
+
+            // Scanning
+            CPubKey sum_input_pub_keys = CPubKey::Combine(input_pub_keys);
+
+            const auto expected_addresses = expected["addresses"].getValues();
+            // We know there is only one address, but if we support labels, this could be multiple addresses
+            CPubKey ecdh_pubkey = ComputeECDHSharedSecret(scan_priv_key, sum_input_pub_keys, HashOutpoints(outpoints));
+            std::vector<uint256> found_tweaks = GetTxOutputTweaks(spend_priv_key.GetPubKey(), ecdh_pubkey, output_pub_keys);
```

Comment:
In 78d1426fabd7b5b6cdd28de508a1692fce291174 "Add BIP352 test vectors as unit tests"

nit: `s/spend_priv_key.GetPubKey()/spend_pubkey`

```suggestion
            std::vector<uint256> found_tweaks = GetTxOutputTweaks(spend_pubkey, ecdh_pubkey, output_pub_keys);
```